File: patterns/bewegungssensor-timer/rules-python/bewegungssensor_timer.py
import threading

from openhab import rule, Registry
from openhab.triggers import ItemStateUpdateTrigger

occupancy_timer = None


def turn_off():
    global occupancy_timer
    Registry.getItem("MotionDetector1").sendCommand("OFF")
    occupancy_timer = None


@rule(triggers=[ItemStateUpdateTrigger("MotionDetector1", "ON")])
class MotionDetector1ReceivedOn:
    def execute(self, module, input):
        global occupancy_timer
        if occupancy_timer is not None:
            occupancy_timer.cancel()

        occupancy_timer = threading.Timer(5 * 60, turn_off)
        occupancy_timer.start()

File: patterns/bewegungssensor-timer/rules-js/bewegungssensor-timer.js
let occupancyTimer = null;

rules.JSRule({
  name: "MotionDetector1 received ON",
  triggers: [triggers.ItemStateUpdateTrigger("MotionDetector1", "ON")],
  execute: (event) => {
    if (occupancyTimer === null || occupancyTimer.hasTerminated()) {
      occupancyTimer = actions.ScriptExecution.createTimer(
        time.ZonedDateTime.now().plusMinutes(5),
        () => {
          items.getItem("MotionDetector1").sendCommand("OFF");
          occupancyTimer = null;
        }
      );
    } else {
      occupancyTimer.reschedule(time.ZonedDateTime.now().plusMinutes(5));
    }
  }
});
